build.sbt

```diff
@@ -342,6 +359,43 @@ lazy val zioHttpGen = (project in file("zio-http-gen"))
   )
   .dependsOn(zioHttpJVM)
 
+lazy val zioHttpGenSbt = (project in file("zio-http-gen-sbt-plugin"))
+  .enablePlugins(SbtPlugin)
+  .settings(publishSetting(true))
+  .settings(
+    name := "zio-http-sbt-codegen",
+    sbtPlugin    := true,
+    scalaVersion := Scala212,
+    semanticdbEnabled := true,
+    semanticdbVersion := scalafixSemanticdb.revision,
+    scalacOptions ++= stdOptions ++ extraOptions(scalaVersion.value),
+    // crossScalaVersions := Seq(Scala212, Scala213),
+    // pluginCrossBuild / sbtVersion := {
+    //   // Not sure why version 1.5.8 specifically is set to minimum version,
+    //   // but when I searched the plugins under sbt org, this seems like the most common:
+    //   // https://github.com/search?q=org%3Asbt%20pluginCrossBuild&type=code
+    //   scalaBinaryVersion.value match {
+    //     case "2.12" => "1.5.8"
+    //     case _      => "2.0.0-M2"
+    //   }
+    // },
+    sbtTestDirectory := sourceDirectory.value / "sbt-test",
+    scriptedLaunchOpts += ("-Dplugin.version=" + version.value),
+    scriptedBufferLog := false,
+    libraryDependencies ++= Seq(
+      `zio-json-yaml`,
+      `zio-test`,
+      `zio-test-sbt`,
+      // when I tried to use the regular .dependsOn(zioHttpGen) syntax,
+      // sbt tried to resolve as an external dependency with current SNAPSHOT version,
+      // and fails (since it is not published).
+      // For now, and until this issue is resolved, we'll just use latest stable published version.
+      //
+      // FIXME: replace the dependency with .dependsOn(zioHttpGen) syntax
+      "dev.zio" %% "zio-http-gen" % "3.0.1",
```

Comment:
I tried, and wasted a few hours trying to figure out why it won't work.
In a clean multi module project, it works.
Only in zio-http repo it acts weird.

similar plugin modules as part of a larger build work in other repos (playframework, tapir, …)

Any idea what's special here?

Reply:
idk
But this is no option, if the plugin is in the same repo. I actually started myself to work on this issue and and problems with the sbt setup. So I created a different repo. Maybe we should do that.

Reply:
Its solveable. same structure works in playframework and tapir projects.
It's also easier to catch integration issues.
If something changes in the `zio-http-gen` module which breaks the plugin, ideally we would want to catch that right away.

I'll try to spend some more time on this :/

Reply:
@987Nabil turns out it works. but only under `++2.12.x`.
Which is expected. When cross version is set to default 2.13, and we try to resolve only 2.12 modules it fails and resorts to trying to download.

So this should work, and we only need carful CI handling for only building/testing the plugin under ++2.12.x